src/Field.tsx

```diff
@@ -83,110 +59,108 @@ export interface FieldConfig {
 
 export type FieldAttributes<T> = GenericFieldHTMLAttributes & FieldConfig & T;
 
-/**
- * Custom Field component for quickly hooking into Formik
- * context and wiring up forms.
- */
-class FieldInner<Values = {}, Props = {}> extends React.Component<
-  FieldAttributes<Props> & { formik: FormikContext<Values> },
-  {}
-> {
-  constructor(
-    props: FieldAttributes<Props> & { formik: FormikContext<Values> }
-  ) {
-    super(props);
-    const { render, children, component } = props;
-    warning(
-      !(component && render),
-      'You should not use <Field component> and <Field render> in the same <Field> component; <Field component> will be ignored'
-    );
-
-    warning(
-      !(component && children && isFunction(children)),
-      'You should not use <Field component> and <Field children> as a function in the same <Field> component; <Field component> will be ignored.'
-    );
+export function useField(name: string, type?: string) {
+  const formik = useFormikContext();
 
+  if (process.env.NODE_ENV !== 'production') {
     warning(
-      !(render && children && !isEmptyChildren(children)),
-      'You should not use <Field render> and <Field children> in the same <Field> component; <Field children> will be ignored'
+      !formik,
+      'useField() / <Field /> must be used underneath a <Formik> component or withFormik() higher order component'
     );
   }
 
-  componentDidMount() {
-    // Register the Field with the parent Formik. Parent will cycle through
-    // registered Field's validate fns right prior to submit
-    this.props.formik.registerField(this.props.name, this);
-  }
+  return formik.getFieldProps(name, type);
+}
```

Comment:
Should `useField` take an optional `validate` function and contain [register logic similar to `Field`](https://github.com/jaredpalmer/formik/blob/hooks/src/Field.tsx#L104-L114)?

Reply:
Yes.